From 0b7cae2fa10da31004a051d8396cbd0e4ceefdfd Mon Sep 17 00:00:00 2001
From: "GL.iNet-XiBao.Lu" <xibao.lu@gl-inet.com>
Date: Wed, 21 Jun 2023 01:23:00 -0700
Subject: [PATCH] dts: fix dts for gl-axt1800

merge patchs:
0005-ipq807x-fit-dts-for-gl-ax-t-1800.patch
0017-gl-axt1800-fix-gl-fan-dts.patch
0024-Add-2-1-usb-bus-for-gl-hw-in-dts.patch

Signed-off-by: GL.iNet-XiBao.Lu <xibao.lu@gl-inet.com>
---
 .../boot/dts/qcom/qcom-ipq6018-gl-axt1800.dts | 41 +++++++++++++++----
 1 file changed, 34 insertions(+), 7 deletions(-)

diff --git a/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-axt1800.dts b/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-axt1800.dts
index 0e5de3f1..05d71d63 100644
--- a/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-axt1800.dts
+++ b/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-axt1800.dts
@@ -24,16 +24,31 @@
 	aliases {
 		sdhc0 = &sdhc_2;
 	};
+
+	gl_fan {
+		compatible = "gl-fan";
+		interrupt-parent = <&tlmm>;
+		interrupts = <31 IRQ_TYPE_EDGE_RISING>;
+	};
+};
+
+&gl_hw {
+	model = "axt1800";
+	lan = "eth1 eth2";
+	usb-port = "1-1,2-1";
+	fan = "hwmon0 cooling_device0";
+	switch-button = "gpio-16";
+	dfs;
 };
 
 &tlmm {
 	sd_pins: sd_pins {
-		sd {
+		mux_0 {
 			pins = "gpio62";
 			function = "sd_card";
 			bias-pull-up;
 		};
-		ldo {
+		mux_1 {
 			pins = "gpio66";
 			function = "gpio";
 			bias-pull-up;
@@ -41,7 +56,7 @@
 	};
 
 	pwm_pins: pwm_pinmux {
-		pwm {
+		mux {
 			pins = "gpio30";
 			function = "pwm13";
 			drive-strength = <8>;
@@ -73,15 +88,27 @@
 		status = "ok";
 	};
 
-	pwm-fan {
+	fan0: pwm-fan {
 		compatible = "pwm-fan";
 		pinctrl-0 = <&fan_pins>;
 		pinctrl-names = "default";
 		cooling-min-state = <0>;
-		cooling-max-state = <3>;
+		cooling-max-state = <255>;
 		#cooling-cells = <2>;
-		pwms = <&pwm 1 255>;
-		cooling-levels = <0 150 200 255>;
+		pwms = <&pwm 1 40000 0>;
+		cooling-levels = <0 36 36 36 36 36 36 36 36 36 36 36 36 36 36 36 36 36 36 36
+				36 36 36 36 36 36 36 36 36 36 36 36 36 36 36 36 36 37 38 39
+				40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59
+				60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79
+				80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99
+				100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119
+				120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139
+				140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159
+				160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179
+				180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199
+				200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219
+				220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239
+				240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255>;
 	};
 };
 
-- 
2.25.1

